week-1-ts-ecosystem/w01e03-object-oriented-structure/src/feature/cart/components/product.ts

```diff
@@ -0,0 +1,21 @@
+interface Product {
+  id: number;
+  amount: number;
+  name: string;
+  price?: number;
+  type: 'Auction' | 'Buy Now' | 'Free';
+}
+
+class Product {
+  constructor(id: number, name: string, price: number, amount: number, type: 'Auction' | 'Buy Now' | 'Free') {
+    this.id = id;
+    this.amount = amount;
+    this.name = name;
+    this.type = type;
+    if (type !== 'Free') {
+      this.price = price;
+    }
+  }
+}
+
+export { Product };
```

Comment:
2 problemy tutaj.

`components` - to powinno być raczej `model` (chodzi o folder)

A 2 sprawa. to po co ten `interface` jeszcze ?

Zobacz, sama klasa `Product` jest jednocześnie kształtem danych i ona wystarczy, `interface` tylko powtarza to co jest zapisane w klasie.

Jest nadmiarowy i niepotrzebny